Add Kubernetes/Helm instructions to the RDS guide

ptgott opened the pull request:
Closes #27753

Add "Kubernetes" tabs as counterparts to Linux host instructions, following the structure we have used in Access Request plugin guides: where we show how to install Teleport on a VM, include `helm repo add` instructions. And when we show how to run a Teleport service, include `helm install` instructions.

In this guide, I've hardcoded a Teleport configuration file in order to run `teleport db configure bootstrap`, then inserted the config file into a Helm values file.

This change also includes instructions to use a local `teleport` binary in order to bootstrap IAM resources.

This also adds minor formatting and readability changes, e.g., the `Var` component.

ptgott committed:
Add Kubernetes/Helm instructions to the RDS guide

Closes #27753

Add "Kubernetes" tabs as counterparts to Linux host instructions,
following the structure we have used in Access Request plugin guides:
where we show how to install Teleport on a VM, include `helm repo add`
instructions. And when we show how to run a Teleport service, include
`helm install` instructions.

In this guide, I've hardcoded a Teleport configuration file in order to
run `teleport db configure bootstrap`, then inserted the config file
into a Helm values file.

This change also includes instructions to use a local `teleport` binary
in order to bootstrap IAM resources.

This also adds minor formatting and readability changes, e.g., the `Var`
component.

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -43,74 +53,294 @@ This guide will help you to:
+`"us-west-1"`) in which you would like the Teleport Database Service to discover
+databases.
+
+The Helm chart that we will  use to install the Teleport Database Service mounts
+a secret containing the join token we generated earlier to
+`/etc/teleport-secrets/auth-token`. The configuration above instructs the
+Database Service to search for its token there.
```
Since we never explicitly create this secret or define a configuration that does, I assume that the helm chart is preset to look for `/tmp/token` and creates the secret if found. But you know what they say about assumptions, so I'd mentioning it.

ptgott committed:
Respond to alexfornuto feedback

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -43,74 +53,296 @@ This guide will help you to:
+
+Later in this guide, you will install a Helm chart to launch the Teleport
+Database Service. The chart creates a secret containing the join token you
+generatd earlier in this guide and mounts it at the path
```
```suggestion
generated earlier in this guide and mounts it at the path
```

ptgott committed:
Update docs/pages/database-access/guides/rds.mdx

Co-authored-by: Alex Fornuto <alex.fornuto@goteleport.com>

ptgott committed:
Fix linter warnings

stevenGravy commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -43,74 +55,296 @@ This guide will help you to:
+
+<Admonition title="Do not do this in production" type="warning">
+
+While we are attaching policies to the EKS node group for simplicity, this
```
We should be using a service account, not attaching to a node role.  See https://github.com/gravitational/teleport/discussions/24116

ptgott replied:
@stevenGravy Is it safe to assume that enough readers will have the OIDC provider installed in their cluster to make this a prerequisite? Otherwise, the extra steps are too ungainly to include in this guide, I think.

stevenGravy replied:
Yes we don't need to go into details on the oidc provider config

ptgott replied:
@stevenGravy I've changed the guide to describe using the OIDC provider, though when I followed these instructions, the Teleport Database Service pod showed this error in its logs: 

```
2023-07-13T19:45:46Z ERRO [IAM]       Failed to auto-configure IAM for Database(Name=example, Type=rds, Labels=map[env:dev teleport.dev/origin:config-file]). error:[
ERROR REPORT:
Original Error: *awserr.baseError WebIdentityErr: failed to retrieve credentials
caused by: AccessDenied: Not authorized to perform sts:AssumeRoleWithWebIdentity
        status code: 403, request id: 6d38d1d1-4f06-4198-afb5-c84502078226
Stack Trace:
        github.com/gravitational/teleport/lib/cloud/aws/identity.go:112 github.com/gravitational/teleport/lib/cloud/aws.GetIdentityWithClient
        github.com/gravitational/teleport/lib/srv/db/cloud/iam.go:218 github.com/gravitational/teleport/lib/srv/db/cloud.(*IAM).getAWSIdentity
        github.com/gravitational/teleport/lib/srv/db/cloud/iam.go:182 github.com/gravitational/teleport/lib/srv/db/cloud.(*IAM).getAWSConfigurator
        github.com/gravitational/teleport/lib/srv/db/cloud/iam.go:249 github.com/gravitational/teleport/lib/srv/db/cloud.(*IAM).processTask
        github.com/gravitational/teleport/lib/srv/db/cloud/iam.go:125 github.com/gravitational/teleport/lib/srv/db/cloud.(*IAM).Start.func1
        runtime/asm_amd64.s:1598 runtime.goexit
User Message: WebIdentityErr: failed to retrieve credentials
caused by: AccessDenied: Not authorized to perform sts:AssumeRoleWithWebIdentity
        status code: 403, request id: 6d38d1d1-4f06-4198-afb5-c84502078226] cloud/iam.go:127
 ```
 
 I checked the Service Account annotation and trust policy, and everything seems to be in order. Is there anything I'm missing? Thanks!

ptgott committed:
Add Service Account IAM instructions

Frame the guide around authenticating the Teleport agent pod's service
account via the IAM OIDC provider.

stevenGravy commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,347 @@ This guide will help you to:
+Create an IAM policy document that allows an IAM identity to connect to your RDS
+database. Assign <Var name="rds-region" /> to the name of the AWS region where
+your RDS database is running, <Var name="aws-account" /> to your AWS account
+number, and <Var name="resource-id" /> to the resource ID of your RDS or Aurora
```
We should mention this is the cluster id for Aurora DBs.

stevenGravy commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,347 @@ This guide will help you to:
+             "rds-db:connect"
+         ],
+         "Resource": [
+             "arn:aws:rds-db:<Var name="rds-region" />:<Var name="aws-account" />:dbuser:<Var name="resource-id" />/*"
```
Instead of variable names in the content I would have these as env variables assigned to the variable names. I'm finding some of these outputs are malformed.

ptgott replied:
If the outputs are malformed, I would treat that as a bug with the `Var` component, which we can fix. What issues are you seeing?

stevenGravy replied:
I'll post an example after I recreate

ptgott replied:
Thanks! We fixed an issue with copying the `Var` component's value recently, so you might need to rebase your local `docs` clone against `main`.

I've also noticed that the trust policy document I included in the draft has some errors in it, so I'll work on fixing that and see if the fixes resolve my issues connecting to my RDS database.

alexfornuto replied:
Alternatively you could use the `var` syntax and just provide the file contents without the `cat` command, giving a little more wiggle room for folks with their own preferred methods of creating/editing files.

stevenGravy commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,347 @@ This guide will help you to:
+    region: "<Var name="rds-region" />"
+annotations:
+  serviceAccount:
+    eks.amazonaws.com/role-arn:
```
This should be one line

ptgott committed:
Respond to stevenGravy feedback

ptgott committed:
Minor tweaks

- Ensure `Var` names are consistent
- Use obviously fake values for resource URIs
- Use the correct value for the OIDC issuer ID

ptgott commented:
@stevenGravy I've responded to your feedback and confirmed that the instructions work!

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,348 @@ This guide will help you to:
+   If you have an IAM OIDC provider associated with your cluster, this command
+   will print its ID.
+
+ - The [`jq` CLI](https://jqlang.github.io/jq/), which we use to process JSON
```
```suggestion
 - The [`jq` CLI tool](https://jqlang.github.io/jq/), which we use to process JSON
```

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,348 @@ This guide will help you to:
+In this section, you will configure the Teleport Database Service. To do so, you
+will:
+
+- Create a join token for the service to demonstrate trust with your Teleport
+  cluster
+- Set up your package manager so you can install and run the Database Service
+- Generate a configuration for the Database Service
```
Suggest commas and/or periods at the end of UL items.

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,348 @@ This guide will help you to:
 
-<ScopedBlock scope={["oss", "enterprise"]}>
+Run the command below on your Linux host. Assign <Var
+name="mytenant.teleport.sh:443" /> to the host **and port** of your Teleport
```
Xin said something in another PR about moving away from the term `tenant`, so you might wanna replace `mytenant` with `example` or similar.



```suggestion
name="example.teleport.sh:443" /> to the host **and port** of your Teleport
```

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,348 @@ This guide will help you to:
 
-</ScopedBlock>
-<ScopedBlock scope={["cloud"]}>
+The command will generate a Database Service configuration with RDS/Aurora
```
```suggestion
The command will generate a Teleport Database Service configuration with RDS/Aurora
```

Also, should it be Service or A/agent?

ptgott replied:
The Core Concepts page says "Teleport Database Service"

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -18,99 +18,348 @@ This guide will help you to:
-Before you can generate IAM permissions, you must provide the Teleport Database
-Service access to AWS credentials.
+In this step, we will show you how to provide the Teleport Database Service
+access to AWS credentials by using the Teleport configuration file we created
+earlier to automatically create and attach IAM policies:
```
Isn't the config file only created when following the "Linux Host" tabs? Wouldn't that make this line confusing to those on the K8S tabs?

ptgott replied:
The second half of this paragraph is a holdover from instructions I added to use `teleport db configure bootstrap` and then deleted. I've made this paragraph more general.

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
````diff
@@ -18,99 +18,348 @@ This guide will help you to:
+Proxy Service, and <Var name="rds-uri" /> to the host **and port** of your RDS
+database (e.g., `myrds.us-east-1.rds.amazonaws.com:5432`):
+
+```code
````
Same comment as above re: `var` syntax and wiggle room.

alexfornuto commented on docs/pages/database-access/guides/rds.mdx:
```diff
@@ -158,52 +405,42 @@ $ tsh db ls
   `<cluster-id>-custom-<endpoint-name>` respectively.
 
-  You can override the `<cluster-id>` part of the name with `TeleportDatabaseName` AWS tag.
+  You can override the `<cluster-id>` part of the name with
```
```suggestion
  You can override the `<cluster-id>` part of the name with the
```

alexfornuto commented on docs/pages/includes/database-access/aws-bootstrap.mdx:
```diff
@@ -64,6 +64,3 @@ You will need to run the bootstrap command once with the Teleport Database
 IAM role that is used for `assume_role_arn`.
 </Details>
-
-See the full `bootstrap` command
-[reference](../../database-access/reference/cli.mdx#teleport-db-configure-bootstrap).
```
Why remove a link to the reference doc?

ptgott replied:
This is to abide by our style guide rule to not include links in the body text unless they're necessary. It's not necessary to follow the reference guide to get through the how-to guides that include this partial.

ptgott committed:
Respond to alexfornuto feedback

ptgott committed:
Fix spelling

ptgott committed:
Fix linter issues
